ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -208,6 +217,37 @@ public KsqlQueryType getQueryType() {
     return KsqlQueryType.PERSISTENT;
   }
 
+  public String getTopologyDescription() {
+    return topology.describe().toString();
+  }
+
+  public List<QueryError> getQueryErrors() {
+    return ImmutableList.copyOf(queryErrors);
+  }
+
+  protected boolean isClosed() {
+    return closed;
+  }
+
+  protected KafkaStreams getKafkaStreams() {
+    return kafkaStreams;
+  }
+
+  protected void resetKafkaStreams(final KafkaStreams kafkaStreams) {
+    this.kafkaStreams = kafkaStreams;
+    queryErrors.clear();
```

Comment:
we don't want to clear errors here, right? Otherwise when in a retry loop we may not actually return any errors back. I think it would be better to accumulate them (up to a certain size/time bound)

Reply:
Done.

I added a change in the `QueryMetadata` to replace the `Set` with an `EvictionQueue` that has a max. size. The max. size is configurable with `ksql.query.error.max.queue.size` (default: 10). 

I'm not sure if we need the `Set` just to avoid repeatable error messages. Perhaps is useful if we keep every error reported after every restart even if they're the same?

Reply:
I agree a Set doesn't make sense. It's useful to be able to see all the errors and when they occurred.